peekaboo/ruleset/rules.py

```diff
@@ -519,23 +519,26 @@ def evaluate(self, sample):
                     break
                 except IdentifierMissingException as missing:
                     identifier = missing.name
-                    if identifier == "cuckooreport":
-                        value = self.get_cuckoo_report(sample)
-                        if value is None:
-                            return self.result(
-                                Result.failed,
-                                _("Evaluation of expression couldn't get cuckoo report."),
-                                False)
-                    elif identifier == "olereport":
-                        value = self.get_oletools_report(sample)
-                    # here elif for other reports
-                    else:
+
+                if identifier == "cuckooreport":
+                    value = self.get_cuckoo_report(sample)
+                    if value is None:
                         return self.result(
                             Result.failed,
-                            _("Evaluation of expression uses undefined identifier."),
+                            _("Evaluation of expression couldn't get cuckoo "
+                              "report."),
                             False)
-
-                    context['variables'][identifier] = value
+                elif identifier == "olereport":
+                    value = self.get_oletools_report(sample)
```

Comment:
I've traced the mentioned lockups to regex matching in olevba's detect_suspicious() (https://github.com/decalage2/oletools/blob/2aa07c1204670b3b42d9b3b566b8e82938f0b897/oletools/olevba.py#L2167). It might have to do with the fact that I've used 512-byte-sized files out of /dev/urandom when trying to reproduce the StopIteration error. olevba most of the time accepts those as valid documents and then seems to choke on them. I've basiclly been fuzzing olevba. :)

Since this only happens in an external component, with invalid input, only with python2 and python2 is dead anyway, I'd not invest any more effort here for the time being.